pkg/cvo/sync_worker.go

```diff
@@ -424,11 +424,8 @@ func (w *SyncWorker) calculateNext(work *SyncWork) bool {
 		work.Attempt = 0
 	}
 
-	if w.work != nil {
```

Comment:
I mean this one
https://github.com/openshift/cluster-version-operator/blob/c56a72560f794012c71faf712258265c41ed1733/pkg/cvo/sync_worker.go#L415

I'm just cleaning up some code scanner results which highlighted this. Strictly speaking this doesn't cause any problems.

Reply:
That's good and what I figured you were working on.

But regardless of what happens in `equalSyncWork` I'm not seeing anything that prevents dereferencing a nil `w.work` [here](https://github.com/openshift/cluster-version-operator/blob/c56a72560f794012c71faf712258265c41ed1733/pkg/cvo/sync_worker.go#L427). Am I missing it?

Reply:
Oh yeah, sorry for all the noise. All the equalSyncWork's nil checking threw me. So there's no need to check for `a == nil` as we would've already blew up but that's a different issue.  I think I've checked this before, i.e. that at this point in the code `w.work` is never nil but again, different (albeit larger) issue.

Reply:
> So `w.work` is already dereferenced when `equalSyncWork` is called. If it's `nil` there's going to be a panic and the execution won't get to the check any way.

I'm not convinced we'd have blown up before.  [Here][1] is where we gracefully exit from `equalSyncWork` if either argument is `nil`.  And if there was a situation where we'd panic, I'd rather address that by fixing the panic, instead of removing code that we wouldn't hit because we panic.  Holding until we bottom this out.

/hold

[1]: https://github.com/openshift/cluster-version-operator/blob/ea6e779ef89710879e2c08c0e5847a5b5e860b28/pkg/cvo/sync_worker.go#L473-L474

Reply:
Yes, agree with @wking. That was my concern to begin with and somehow I lost the handle in all the back-and-forth. We still need that check.

Reply:
The code looks confusing to me. I will go through it and utilize this opportunity to simplify it.